nuxeo-connect-client/src/main/java/org/nuxeo/connect/packages/PackageManagerImpl.java

```diff
@@ -427,6 +426,30 @@ public void registerSource(PackageSource source, boolean local) {
     }
 
     @Override
+    public List<String> listInstalledPackagesNames(PackageType pkgType) {
+        List<DownloadablePackage> upgrades = listInstalledPackages();
+        List<String> upgradeNames = new ArrayList<>();
+        for (DownloadablePackage upgrade : upgrades) {
+            if (pkgType == null || upgrade.getType() == pkgType) {
+                upgradeNames.add(upgrade.getName());
+            }
+        }
+        return upgradeNames;
+    }
+
+    @Override
+    public List<String> listHotfixesNames(String targetPlatform, boolean allowSNAPSHOT) {
+        List<DownloadablePackage> hotFixes = listPackages(PackageType.HOT_FIX, targetPlatform);
+        List<String> hotFixNames = new ArrayList<>();
+        hotFixes.forEach(pkg -> {
+            if(!hotFixNames.contains(pkg.getName()) && (!pkg.getVersion().isSnapshot() || allowSNAPSHOT)){
+                hotFixNames.add(pkg.getName());
```

Comment:
Here, it's better to use stream().filter(pkg -> ...).map(...).collect(Collectors.toList()); as it there's no possible side effect as you construct the new list and not modify an external one. (http://stackoverflow.com/questions/28319064/java-8-best-way-to-transform-a-list-map-or-foreach)